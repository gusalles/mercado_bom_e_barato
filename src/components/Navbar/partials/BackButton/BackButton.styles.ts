import { styled } from '@mui/material';

export const Button = styled('button')(() => ({
  backgroundColor: 'transparent',
  border: 0,
  cursor: 'pointer',
}));
