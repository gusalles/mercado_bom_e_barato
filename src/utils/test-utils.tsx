import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
