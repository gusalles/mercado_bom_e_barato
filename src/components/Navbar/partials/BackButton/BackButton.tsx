import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import { Button } from './BackButton.styles';

interface BackButtonProps {
  onClick: () => void;
}

export function BackButton({ onClick }: BackButtonProps) {
  return (
    <Button type="button" onClick={onClick}>
      <ArrowBackIcon color="secondary" />
    </Button>
  );
}
