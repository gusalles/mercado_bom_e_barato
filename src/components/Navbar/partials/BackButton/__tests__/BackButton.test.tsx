import { render, screen, userEvent } from '@/utils/test-utils';
import { BackButton } from '../BackButton';

describe('BackButton', () => {
  const mockClickTrigger = vi.fn();

  it('should render', () => {
    render(<BackButton onClick={mockClickTrigger} />);

    const button = screen.getByRole('button');

    expect(button).toMatchSnapshot();
  });

  it('should trigger the click effect', async () => {
    render(<BackButton onClick={mockClickTrigger} />);

    const button = screen.getByRole('button');

    await userEvent.click(button);

    expect(mockClickTrigger).toBeCalled();
  });
});
