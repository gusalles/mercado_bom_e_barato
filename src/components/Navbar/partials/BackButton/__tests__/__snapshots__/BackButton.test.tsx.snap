// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackButton > should render 1`] = `
<button
  class="css-i5nnfd"
  type="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-25to7p-MuiSvgIcon-root"
    data-testid="ArrowBackIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
    />
  </svg>
</button>
`;
